compiler/rustc_typeck/src/check/inherited.rs

```diff
@@ -161,7 +161,23 @@ impl Inherited<'a, 'tcx> {
     where
         T: TypeFoldable<'tcx>,
     {
-        let ok = self.partially_normalize_associated_types_in(span, body_id, param_env, value);
+        self.normalize_associated_types_in_with_cause(
+            ObligationCause::misc(span, body_id),
+            param_env,
+            value,
+        )
+    }
+
+    pub(super) fn normalize_associated_types_in_with_cause<T>(
```

Comment:
I'm not a big fan of creating multiple new functions that are variations of each other. It makes it confusing which to use, why, and when in the future. Might just make sense to pass an `ObligationCause` everywhere...

Reply:
I wanted to avoid a large amount of churn. I may do a follow-up PR to further investigate the impact of the unifying these functions.